app/services/cart.js

```diff
@@ -1,20 +1,35 @@
 import Ember from 'ember';
 
 export default Ember.Service.extend({
-  addToCart(title, author, price) {
-    //get the existing cart and add the book, or if there's not a cart then create a new cart with the book in it
-    let cart = JSON.parse(localStorage.getItem('cart'));
-    if (!cart) {
-      cart = [];
-    } 
-    cart.push({ title, author, price });
-    localStorage.setItem('cart', JSON.stringify(cart));
 
+  init() {
+  // get the cart our of local storage (this way it persists in the user's browser past a reload)
+  this.set('cart', JSON.parse(localStorage.getItem("cart")));
+  if (!this.cart) { 
+    this.set('cart',[]); 
+  }
+  // send the current total in the cart to the template as well
+  this.set('total',this._calcTotal());
+  },
+
+  //add an item to the cart in both local storage and the current state
+  addToCart(title, author, price) {
+    this.cart.pushObject({ title, author, price });
+    localStorage.setItem('cart', JSON.stringify(this.cart));
+    this.set('total', this._calcTotal());
   },
 
+  //clear out the cart in local storage and in state
   emptyCart() {
-    localStorage.removeItem('cart');
-    window.location.reload();
-  }
+    localStorage.setItem('cart', '[]');
+    this.set('cart',[]);
```

Comment:
we love consistency here at Smile....making sure there's an extra space after a `,` in a list of arguments goes a long way